store/copr/mpp.go

```diff
@@ -260,13 +295,21 @@ func (m *mppIterator) establishMPPConns(bo *tikv.Backoffer, req *kv.MPPDispatchR
 		return
 	}
 
-	// TODO: cancel the whole process when some error happens
 	for {
 		err := m.handleMPPStreamResponse(bo, resp, req)
 		if err != nil {
+			m.cancelMppTasks(bo, taskMeta)
 			m.sendError(err)
 			return
 		}
+
+		if m.vars != nil && m.vars.Killed != nil && atomic.LoadUint32(m.vars.Killed) == 1 {
```

Comment:
Checking killed status **before Recv** is pointless. Because once the kill is launched, the Recv function will react and return the error as soon as possible.

Reply:
if the cancel of Recv is not called immediately or out of exception, we should  ensure this goroutine will be killed.